src/transaction/transaction.rs

```diff
@@ -784,11 +789,10 @@ impl<PdC: PdClient> Transaction<PdC> {
                         break;
                     }
                 }
-                let current_ts = rpc.clone().get_timestamp().await?;
                 let request = new_heart_beat_request(
                     start_ts.clone(),
                     primary_key.clone(),
-                    (current_ts.physical - start_ts.physical) as u64 + DEFAULT_LOCK_TTL,
+                    elapsed + DEFAULT_LOCK_TTL,
```

Comment:
`elapaed` will not be updated during iterations, you need to move the instant into the closure and calculate elapsed() each time.